pkg/trace/stats/concentrator.go

```diff
@@ -215,6 +228,59 @@ func (c *Concentrator) Add(t Input) {
 	c.mu.Unlock()
 }
 
+// ProcessOTLPTraces applies APM stats calculation on the otlp traces in the concentrator.
+// This function is NOT called in Concentrator.Run(), so you need to manually call it to get APM stats on the given OTLP traces.
+func (c *Concentrator) ProcessOTLPTraces(traces ptrace.Traces, additionalCtagKeys ...string) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	includeCID := shouldIncludeCIDDim(c.conf)
+	spanByID, resByID, scopeByID := traceutil.IndexOTelSpans(traces)
+	topLevelByKind := c.conf.HasFeature("enable_otlp_compute_top_level_by_span_kind")
+	topLevelSpans := traceutil.GetTopLevelOTelSpans(spanByID, resByID, topLevelByKind)
+	ignoreResNames := make(map[string]struct{})
+	for _, resName := range c.conf.Ignore["resource"] {
+		ignoreResNames[resName] = struct{}{}
+	}
+	for spanID, span := range spanByID {
+		res := resByID[spanID]
+		resName := traceutil.GetOTelResource(span, res)
```

Comment:
qq, does this handle one resource having multiple Spans ?

Reply:
Yes, I added a test